sdk/messaging/azservicebus/client.go

```diff
@@ -14,23 +14,30 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus/internal"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus/internal/utils"
 	"github.com/devigned/tab"
 )
 
 // Client provides methods to create Sender and Receiver
 // instances to send and receive messages from Service Bus.
 type Client struct {
+	// NOTE: values need to be 64-bit aligned. Simplest way to make sure this happens
+	// is just to make it the first value in the struct
+	// See:
+	//   Godoc: https://pkg.go.dev/sync/atomic#pkg-note-BUG
+	//   PR: https://github.com/Azure/azure-sdk-for-go/pull/16847
 	linkCounter uint64
-	links       map[uint64]internal.Closeable
-	creds       clientCreds
-	namespace   interface {
+
+	linksMu   *sync.Mutex
```

Comment:
The ordering of these changed, was that intentional?

Reply:
Just reordering for it be logical and to keep the mutex near the stuff it actually protects (it's not perfect, but the alignment requirement makes it less easy)